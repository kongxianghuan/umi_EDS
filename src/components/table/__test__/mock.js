export const data = [
  { key: 1, name: 'ben', age: 41, address: 'guangdong', country: 'China' },
  { key: 2, name: 'doge', age: 21, address: 'london', country: 'UK' },
  { key: 3, name: 'aya', age: 31, address: 'shanxi', country: 'China' },
  { key: 4, name: 'cat', age: 11, address: 'newyork', country: 'USA' },
  { key: 5, name: 'jack', age: 24, address: 'shenzhen', country: 'China' }
]

export const columns = [
  { key: 1, dataIndex: 'name', title: 'name' },
  { key: 2, dataIndex: 'age', title: 'age' },
  { key: 3, dataIndex: 'address', title: 'address' },
  { key: 4, dataIndex: 'country', title: 'country' },
  {
    key: 5, dataIndex: 'action', title: 'action',
    render: (text, record) => <button>click me</button>
  },
]

export const pagingData = [
  { key: 1, name: 'ben', age: 41, address: 'guangdong', country: 'china' },
  { key: 2, name: 'doge', age: 21, address: 'london', country: 'UK' },
  { key: 3, name: 'aya', age: 31, address: 'shanxi', country: 'china' },
  { key: 4, name: 'cat', age: 11, address: 'newyork', country: 'USA' },
  { key: 5, name: 'jack', age: 24, address: 'shenzhen', country: 'china' },
  { key: 11, name: 'ben', age: 41, address: 'guangdong', country: 'china' },
  { key: 12, name: 'doge', age: 21, address: 'london', country: 'UK' },
  { key: 13, name: 'aya', age: 31, address: 'shanxi', country: 'china' },
  { key: 14, name: 'cat', age: 11, address: 'newyork', country: 'USA' },
  { key: 15, name: 'jack', age: 24, address: 'shenzhen', country: 'china' },
  { key: 21, name: 'ben', age: 41, address: 'guangdong', country: 'china' },
  { key: 22, name: 'doge', age: 21, address: 'london', country: 'UK' },
  { key: 23, name: 'aya', age: 31, address: 'shanxi', country: 'china' },
  { key: 24, name: 'cat', age: 11, address: 'newyork', country: 'USA' },
  { key: 25, name: 'jack', age: 24, address: 'shenzhen', country: 'china' },
  { key: 31, name: 'ben', age: 41, address: 'guangdong', country: 'china' },
  { key: 32, name: 'doge', age: 21, address: 'london', country: 'UK' },
  { key: 33, name: 'aya', age: 31, address: 'shanxi', country: 'china' },
  { key: 34, name: 'cat', age: 11, address: 'newyork', country: 'USA' },
  { key: 35, name: 'jack', age: 24, address: 'shenzhen', country: 'china' },
  { key: 41, name: 'ben', age: 41, address: 'guangdong', country: 'china' },
  { key: 42, name: 'doge', age: 21, address: 'london', country: 'UK' },
  { key: 43, name: 'aya', age: 31, address: 'shanxi', country: 'china' },
  { key: 44, name: 'cat', age: 11, address: 'newyork', country: 'USA' },
  { key: 45, name: 'jack', age: 24, address: 'shenzhen', country: 'china' },
  { key: 51, name: 'ben', age: 41, address: 'guangdong', country: 'china' },
  { key: 52, name: 'doge', age: 21, address: 'london', country: 'UK' },
  { key: 53, name: 'aya', age: 31, address: 'shanxi', country: 'china' },
  { key: 54, name: 'cat', age: 11, address: 'newyork', country: 'USA' },
  { key: 55, name: 'jack', age: 24, address: 'shenzhen', country: 'china' },
  { key: 61, name: 'ben', age: 41, address: 'guangdong', country: 'china' },
  { key: 62, name: 'doge', age: 21, address: 'london', country: 'UK' },
  { key: 63, name: 'aya', age: 31, address: 'shanxi', country: 'china' },
  { key: 64, name: 'cat', age: 11, address: 'newyork', country: 'USA' },
  { key: 65, name: 'jack', age: 24, address: 'shenzhen', country: 'china' },
  { key: 71, name: 'ben', age: 41, address: 'guangdong', country: 'china' },
  { key: 72, name: 'doge', age: 21, address: 'london', country: 'UK' },
  { key: 73, name: 'aya', age: 31, address: 'shanxi', country: 'china' },
  { key: 74, name: 'cat', age: 11, address: 'newyork', country: 'USA' },
  { key: 75, name: 'jack', age: 24, address: 'shenzhen', country: 'china' },
]