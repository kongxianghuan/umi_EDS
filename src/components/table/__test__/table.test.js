import React from 'react'
import { shallow } from 'enzyme'
import Table from '..'
import { columns, data } from './mock'

describe('Table', () => {
  it('should render correctly', () => {
    const table = shallow(<Table columns={columns} data={data} />)
    expect(table).toMatchSnapshot()
  }) 
})
