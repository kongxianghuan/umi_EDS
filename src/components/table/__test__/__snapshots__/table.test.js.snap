// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render correctly 1`] = `
<div
  className="eds-table"
>
  <div
    className="eds-table-loading hidden"
  >
    <Loading
      size="medium"
      type="div"
    />
  </div>
  <table
    className="table "
  >
    <thead>
      <tr>
        <th
          className={Array []}
          key="1"
          onClick={null}
        >
          name
        </th>
        <th
          className={Array []}
          key="2"
          onClick={null}
        >
          age
        </th>
        <th
          className={Array []}
          key="3"
          onClick={null}
        >
          address
        </th>
        <th
          className={Array []}
          key="4"
          onClick={null}
        >
          country
        </th>
        <th
          className={Array []}
          key="5"
          onClick={null}
        >
          action
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className=""
        key="1"
      >
        <td
          key="1"
        >
          ben
        </td>
        <td
          key="2"
        >
          41
        </td>
        <td
          key="3"
        >
          guangdong
        </td>
        <td
          key="4"
        >
          China
        </td>
        <td
          key="5"
        >
          <button>
            click me
          </button>
        </td>
      </tr>
      <tr
        className=""
        key="2"
      >
        <td
          key="1"
        >
          doge
        </td>
        <td
          key="2"
        >
          21
        </td>
        <td
          key="3"
        >
          london
        </td>
        <td
          key="4"
        >
          UK
        </td>
        <td
          key="5"
        >
          <button>
            click me
          </button>
        </td>
      </tr>
      <tr
        className=""
        key="3"
      >
        <td
          key="1"
        >
          aya
        </td>
        <td
          key="2"
        >
          31
        </td>
        <td
          key="3"
        >
          shanxi
        </td>
        <td
          key="4"
        >
          China
        </td>
        <td
          key="5"
        >
          <button>
            click me
          </button>
        </td>
      </tr>
      <tr
        className=""
        key="4"
      >
        <td
          key="1"
        >
          cat
        </td>
        <td
          key="2"
        >
          11
        </td>
        <td
          key="3"
        >
          newyork
        </td>
        <td
          key="4"
        >
          USA
        </td>
        <td
          key="5"
        >
          <button>
            click me
          </button>
        </td>
      </tr>
      <tr
        className=""
        key="5"
      >
        <td
          key="1"
        >
          jack
        </td>
        <td
          key="2"
        >
          24
        </td>
        <td
          key="3"
        >
          shenzhen
        </td>
        <td
          key="4"
        >
          China
        </td>
        <td
          key="5"
        >
          <button>
            click me
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
